### ISSUE TITLE
coroutines-guide.md: usages of delay(Long) need to be replaced with delay(Int).

### ISSUE THREAD

**Issue by hastebrot** [Kotlin/kotlinx.coroutines#392]
As of version 0.23.0 `delay()` has the following signatures:

- `delay(Long, TimeUnit)`
- `delay(Int)`

I assume we just need to replace all usages to use `delay(Int)`.

**Comment by hastebrot** [comment]
Same for `withTimeoutOfNull()`.

**Comment by qwwdfsad** [comment]
No, both version should be present.

`Long` version is required on JVM for granular delays (e.g. nanosecond scale), while `Int` is useful on JS where `Long` is always a boxed object

**Comment by hastebrot** [comment]
@qwwdfsad So does the example in https://github.com/Kotlin/kotlinx.coroutines/blob/master/coroutines-guide.md#your-first-coroutine still work?

~~~kt
fun main(args: Array<String>) {
    launch { // launch new coroutine in background and continue
        delay(1000L) // non-blocking delay for 1 second (default time unit is ms)
        println("World!") // print after delay
    }
    println("Hello,") // main thread continues while coroutine is delayed
    Thread.sleep(2000L) // block main thread for 2 seconds to keep JVM alive
}
~~~

**Comment by qwwdfsad** [comment]
It is.
In my env (IDEA 2018.1, Kotlin plugin 1.60-dev) I don't see any warnings about resolution ambiguity, could you please provide more details?

**Comment by paolop** [comment]
Intellij 2018.1 after updating to 0.23.0 warns about resolution ambiguity of the delay method (delay(333L)) - note the Long actual parameter. The code compiles correctly anyway.Il 10 giu 2018 12:28, Vsevolod Tolstopyatov <notifications@github.com> ha scritto:It is.
In my env (IDEA 2018.1, Kotlin plugin 1.60-dev) I don't see any warnings about resolution ambiguity, could you please provide more details?

**Comment by hastebrot** [comment]
I'm on Kotlin 1.2.41 (i.e. latest stable).

**Comment by hastebrot** [comment]
This has been fixed with Kotlin 1.2.50.

It now resolves the call `delay(1000L)` to

~~~kt
public suspend fun delay(time: Long, unit: TimeUnit = TimeUnit.MILLISECONDS)
~~~

Closing this issue.
